src/core/client.ts

```diff
@@ -98,6 +98,16 @@ export enum DocumentSyncResultType {
   SyncFailed = 'sync-failed',
 }
 
+/**
+ * `PeersChangedValue` represents the value of the PeersChanged event.
+ * @public
+ */
+export type PeersChangedValue<P> = {
+  type: 'initialization' | 'watched' | 'unwatched' | 'presence-changed';
+  changedPeers: Record<DocumentKey, Array<{ clientID: ActorID; presence: P }>>;
+  peers: Record<DocumentKey, Array<{ clientID: ActorID; presence: P }>>;
```

Comment:
Do we need both `changedPeers` and `peers`? I wonder if `peers` can be used together depending on the event type. 🤔

Reply:
`PeersChangedValue` is a value of the `peers-changed` event, 
which contains the entire peer information in `peers` 
and additional information about the changed peer in `changedPeers`.
This is different from the previous event which only sent the whole peer information.

It contains `peers` in case there is a need to know the entire peers for the cases of update and delete. 
Would it be sufficient to provide only the `changedPeers` instead of the entire `peers`?

```ts
client.subscribe((event) => {
  switch (event.type) {
    case 'status-changed': {
      // event.value: activated, deactivated
      break;
    }
    case 'stream-connection-status-changed': {
      // event.value: connected, disconnected
      break;
    }
    case 'document-synced': {
      // event.value: synced, sync-failed
      break;
    }
    case 'document-changed': {
      // event.value: docKey[]
      break;
    }
    case 'peers-changed': {
      const { type, changedPeers, peers } = event.value;
      // type: initialization, watched, unwatched, presence-changed

      // peers corresponding to the current document
      const docChangedPeers = changedPeers[doc.getKey()];
      const docPeers = peers[doc.getKey()];
      switch (event.value.type) {
        case 'initialization': {
          displayCursor(docPeers);
          break;
        }
        case 'unwatched': {
          // we can get which peer has unwatched.
          removeCursor(docChangedPeers);
          break;
        }
        case 'watched':
        case 'presence-changed': {
          // we can partially update for changed peers.
          updateCursor(docChangedPeers);
          break;
        }
      }
      break;
    }
  }
});
```

Reply:
I wonder if it is okay to return entire peers as `peers` in initialization and return changed peers as `peers` in `watched` and `unwatched`.

This is because the more variables the more complex it is.